src/components/common/buttons/IconButton.js

```diff
@@ -24,7 +24,7 @@ type IconProps = {
  * @returns {React.Node}
  */
 const IconButton = ({ styles, theme, text, onPress, disabled, name, ...iconProps }: IconProps) => {
-  const _onPress = useOnPress(disabled ? undefined : onPress, [disabled])
+  const _onPress = useOnPress(disabled ? undefined : onPress, [disabled, onPress])
```

Comment:
1. i think it's not needed. just const _onPress = useOnPress(onPress)
2. the function itself adds as the deps **inside** useOnPress hook so it could be omited:
https://github.com/GoodDollar/GoodDAPP/blob/react-native-useOnPress/src/lib/hooks/useOnPress.js#L12
3. i saw this in a few places, please fix everywhere

Reply:
I'm confused with the advice 2, never should we pass a dependency array?

Reply:
yes but
a) disabled ? undefined : onPress is extra logic. it doesn't needed as disabled control won't emit onPress. So just useOnPress(onPress)
b) my hook useOnPress auto adds the callback we're wrapping as the dependency. so you don't need to pass onPress. 

as a result, in this particular case the deps array is empty. so it should be const _onPress = useOnPress(onPress, []) that could be written shortly as _onPress = useOnPress(onPress)

Reply:
done